Remove Quirk shouldAdvertiseSupportForHLSSubtitleTypes on hulu.com

karlcow opened the pull request:
#### 44e22d905dd0893f661a57fed60f98c994f3e809
<pre>
Remove Quirk shouldAdvertiseSupportForHLSSubtitleTypes on hulu.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=254638">https://bugs.webkit.org/show_bug.cgi?id=254638</a>
rdar://107347115

Reviewed by Brent Fulgham.

Enabling/Disabling the Quirks doesn&apos;t change the behavior of hulu.com.
The site plays videos normally. In the current circumstances the Quirk
seems not to be necessary anymore. It might be the results made in
rdar://110588102 too.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::canPlayType const):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAdvertiseSupportForHLSSubtitleTypes const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: <a href="https://commits.webkit.org/266925@main">https://commits.webkit.org/266925@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a7778dc0d25d3192729b1baeb31a789f44f7d12

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14965 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15269 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15623 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16718 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14079 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15103 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17790 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15370 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16710 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15145 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15621 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12715 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17447 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12901 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13501 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20463 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13979 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13669 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16904 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14224 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12029 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13507 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3644 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17840 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14066 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (commented):
I support removing this Quirk, but we need @jernoble 's approval, since it impacts a media streaming site.

brentfulgham reviewed (approved):
I discussed with Jer, and this is safe to remove. We ended up adopting new media APIs that allow us to avoid this hack. r=me

karlcow committed:
Remove Quirk shouldAdvertiseSupportForHLSSubtitleTypes on hulu.com
https://bugs.webkit.org/show_bug.cgi?id=254638
rdar://107347115

Reviewed by Brent Fulgham.

Enabling/Disabling the Quirks doesn't change the behavior of hulu.com.
The site plays videos normally. In the current circumstances the Quirk
seems not to be necessary anymore. It might be the results made in
rdar://110588102 too.

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::canPlayType const):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAdvertiseSupportForHLSSubtitleTypes const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: https://commits.webkit.org/266925@main
